src/component/RecipeList.js

```diff
@@ -0,0 +1,39 @@
+import { useState } from "react";
+import RecipeListItem from "./RecipeListItem";
+import Pagination from "./Pagination";
+
+function RecipeList({ recipes, handleChooseRecipe }) {
+  const [currentPage, setCurrentPage] = useState(1);
+  //   const [recipeList, setRecipeList] = useState([
+  //     ...recipes.slice(startIndex, endIndex),
+  //   ]);
+  const [recordsPerPage] = useState(10);
+
+  const endIndex = currentPage * recordsPerPage;
```

Comment:
You could refactor this a bit for better readability and easier to understand:
```
const getRecipeListByCurrentPageNumber = useCallback(() => {
  const endIndex = currentPage * recordsPerPage;
  const startIndex = endIndex - recordsPerPage;
  return recipes.slice(startIndex, endIndex);
}, [currentPage]);

const renderedRecipes = getRecipeListFromPageNumber().map(...)
```

Reply:
Is this function being passed down to another component, that we have to wrap it in a `useCallback` to avoid unnecessary re-rendering? Or is it a practice to prevent functions being re-created unnecessarily?